packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -461,431 +499,857 @@ describe('Testing authentication module', () => {
+          framelessPostMock.initializeWithContext(context).then(() => {
+            const successCallback = (): void => {
+              done();
+            };
+            const failureCallback = (reason: string): void => {
+              expect(reason).toBe(mockResult);
+              done();
+            };
+            const userRequest: authentication.UserRequest = {
+              successCallback: successCallback,
+              failureCallback: failureCallback,
+            };
+            authentication.getUser(userRequest);
+            const message = framelessPostMock.findMessageByFunc('authentication.getUser');
+            expect(message).not.toBeNull();
+            expect(message.id).toBe(1);
+            expect(message.args[0]).toBe(undefined);
+            framelessPostMock.respondToMessage({
+              data: {
+                id: message.id,
+                args: [false, mockResult],
+              },
+            } as DOMMessageEvent);
+          });
+        });
+
+        it(`authentication.getUser should throw error in getting user profile with ${context} context`, async () => {
+          await framelessPostMock.initializeWithContext(context);
+          const promise = authentication.getUser();
+          const message = framelessPostMock.findMessageByFunc('authentication.getUser');
+          expect(message).not.toBeNull();
+          expect(message.id).toBe(1);
+          expect(message.args[0]).toBe(undefined);
+          framelessPostMock.respondToMessage({
+            data: {
+              id: message.id,
+              args: [false, mockResult],
+            },
+          } as DOMMessageEvent);
+          await expect(promise).rejects.toThrowError(mockResult);
+        });
+
+        it(`authentication.getUser should successfully get user profile with ${context} context`, async () => {
+          await framelessPostMock.initializeWithContext(context);
+          const promise = authentication.getUser();
+          const message = framelessPostMock.findMessageByFunc('authentication.getUser');
+          expect(message).not.toBeNull();
+          expect(message.id).toBe(1);
+          expect(message.args[0]).toBe(undefined);
+          framelessPostMock.respondToMessage({
+            data: {
+              id: message.id,
+              args: [false, mockResult],
+            },
+          } as DOMMessageEvent);
+          await expect(promise).rejects.toThrow(mockResult);
+        });
+      });
+    });
+
+    describe('Testing authentication.notifySuccess function', () => {
+      const allowedContexts = [FrameContexts.authentication];
+
+      it('authentication.notifySuccess should not allow calls before initialization', () => {
+        expect(() => authentication.notifySuccess()).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
```

Comment:
Do you need some positive tests? Aka notify success works when using this set of frame contexts?

Reply:
Yes. I opened a bug to track adding these after GA. I couldn't get the framelessPostMock working as per our earlier conversation. :)

Reply:
Yes. I opened a bug to track adding these after GA. I couldn't get the framelessPostMock working as per our earlier conversation. :)